src/communication/xcomponentPublisher.js

```diff
@@ -0,0 +1,28 @@
+define(function () {
```

Comment:
A better naming of this class xcomponentWebsocketPublisher
because it's an implementation based on WS.
We can imagine other implementation later on (such as Rabbitmq in nodejs)